src/Microsoft.Unity.Analyzers/UnusedCoroutineReturnValue.cs

```diff
@@ -0,0 +1,123 @@
+
+		private static readonly Type coroutineReturnValue = typeof(System.Collections.IEnumerable);
+
+		public override void Initialize(AnalysisContext context)
+		{
+			context.EnableConcurrentExecution();
+			context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+			context.RegisterSyntaxNodeAction(AnalyzeInvocation, SyntaxKind.InvocationExpression);
+		}
+
+		private static void AnalyzeInvocation(SyntaxNodeAnalysisContext context)
+		{
+			var typeInfo = context.Compilation.GetTypeByMetadataName("System.Collections.IEnumerator");
+			
+			var invocation = (InvocationExpressionSyntax)context.Node;
+			var symbol = context.SemanticModel.GetSymbolInfo(invocation);
+			if (symbol.Symbol == null)
+				return;
+
+			if (!IsValidCoroutine(symbol.Symbol, typeInfo, out var methodName))
+				return;
+
+			if (!invocation.Parent.IsKind(SyntaxKind.ExpressionStatement))
+				return;
+
+			context.ReportDiagnostic(Diagnostic.Create(Rule, invocation.GetLocation(), methodName));
+		}
+
+		private static bool IsValidCoroutine(ISymbol symbol, INamedTypeSymbol typeInfo, out string methodName)
+		{
+			methodName = null;
+			if (!(symbol is IMethodSymbol method))
+				return false;
+
+			var containingType = method.ContainingType;
+			if (!containingType.Extends(typeof(UnityEngine.MonoBehaviour)))
+				return false;
+
+			if (!Equals(method.ReturnType, typeInfo))
+				return false;
+
+			methodName = method.Name;
+			return true;
+		}
+	}
+
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class UnusedCoroutineReturnValueCodeFix : CodeFixProvider
+	{
+		public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(UnusedCoroutineReturnValueAnalyzer.Id);
+
+		public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			if (!(root.FindNode(context.Span) is InvocationExpressionSyntax invocation))
+				return;
+
+			var parent = invocation.Parent;
+			if (!parent.IsKind(SyntaxKind.ExpressionStatement))
+				return;
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					Strings.UnusedCoroutineReturnValueCodeFixTitle,
+					ct => WrapWithStartCoroutine(context.Document, parent, invocation, ct),
+					invocation.ToFullString()),
+				context.Diagnostics);
+		}
+
+		private static async Task<Document> WrapWithStartCoroutine(Document document, 
+			SyntaxNode parent, 
+			InvocationExpressionSyntax invocation, 
+			CancellationToken cancellationToken)
+		{
+			var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			var newExpressionStatement = ExpressionStatement(
+				InvocationExpression(
+					IdentifierName("StartCoroutine"), 
+					ArgumentList(
+						SingletonSeparatedList<ArgumentSyntax>(
```

Comment:
<ArgumentSyntax> will be inferred.
`
			var newExpressionStatement = ExpressionStatement(
				InvocationExpression(
					IdentifierName("StartCoroutine"), 
					ArgumentList(SingletonSeparatedList(Argument(invocation)))));
`